### ISSUE TITLE
TLI=1 and RMSEA = 0 for ill-fitting multi-group model

### ISSUE THREAD

**Issue by tbates** [OpenMx/OpenMx#221]
A bad fitting supermodel appears to fit well...

Build a supermodel of vehicle weight in `mtcars` data, with sub-groups for `auto` and `manual` transmissions:

```R
m1 = mxModel("indep", mxFitFunctionMultigroup(c("auto", "man")), 
 mxModel("auto", type = "RAM",  manifestVars = "wt",
	mxPath(labels = "var_wt_grp_auto", from = "wt" , to = "wt", arrows = 2, free = TRUE, values = 1),
	mxPath(labels = "mean_wt_grp_auto", from = "one", to = "wt", arrows = 1, free = TRUE, values = 1),
	mxData(subset(mtcars, am == 0), type = "raw")
 ),
 mxModel("man", type = "RAM",  manifestVars = "wt",
	mxPath(labels = "var_wt_grp_man", from = "wt" , to = "wt", arrows = 2, free = TRUE, values = 1),
	mxPath(labels = "mean_wt_grp_man", from = "one", to = "wt", arrows = 1, free = TRUE, values = 1),
	data = mxData(subset(mtcars, am == 1), type = "raw")
 )
)
m1 = mxRun(m1)
```

And make a comparison model, `m2`, equating the two groups:

```R
m2= umxModify(m1, regex="auto|man", newlabels="eq", name="equated")
```

Of course, `m1` with distinct var and mean for each group is massively better than `m2` where they are set equal:

```R
umxCompare(m1, m2)
```

|Model      | EP|∆ -2LL     |∆ df |p       |      AIC|Compare with |
|:----------|--:|:----------|:----|:-------|--------:|:------------|
|superModel |  4|           |     |        | 10.62001|             |
|equated    |  2|21.7823024 |2    |< 0.001 | 28.40232|superModel   |

But, `summary(m1)` gives no indication of bad fit, saying TLI is 1 and RMSEA is 0... seems incorrect?

> Model Statistics: 
               |  Parameters  |  Degrees of Freedom  |  Fit (-2lnL units)
       Model:              4                     28              66.62001
   Saturated:             NA                     NA                    NA
Independence:             NA                     NA                    NA
Number of observations/statistics: 32/32

> Information Criteria: 
      |  df Penalty  |  Parameters Penalty  |  Sample-Size Adjusted
AIC:       10.62001               74.62001                 76.10150
BIC:      -30.42059               80.48296                 68.01324
CFI: NA 
TLI: 1   (also known as NNFI) 
RMSEA:  0  [95% CI (NA, NA)]
Prob(RMSEA <= 0.05): NA

**Comment by tbates** [comment]
And offering up ref models for this bad-fitting supermodel gives CFI -0.83 (!) while again yielding perfect TLI and RMSEA... (and RMSEA CI and p are not computed...)

```R
m1Sat = mxRefModels(m1, run=TRUE)
summary(m1, refModels=m1Sat)
```
 <pre>
free parameters:
              name matrix row col  Estimate Std.Error A
1  var_wt_grp_auto auto.S  wt  wt 0.5725430 0.1857571  
2 mean_wt_grp_auto auto.M   1  wt 3.7688947 0.1735910  
3   var_wt_grp_man  man.S  wt  wt 0.3513842 0.1378241  
4  mean_wt_grp_man  man.M   1  wt 2.4110000 0.1644066  

Model Statistics: 
               |  Parameters  |  Degrees of Freedom  |  Fit (-2lnL units)
       Model:              4                     28              66.62001
   Saturated:              4                     28              66.62001
Independence:              4                     28              66.62001
Number of observations/statistics: 32/32

chi-square:  χ² ( df=0 ) = -3.126388e-13,  p = 1
Information Criteria: 
      |  df Penalty  |  Parameters Penalty  |  Sample-Size Adjusted
AIC:       10.62001               74.62001                 76.10150
BIC:      -30.42059               80.48296                 68.01324
CFI: -0.8333333 
TLI: 1   (also known as NNFI) 
RMSEA:  0  [95% CI (NA, NA)]
Prob(RMSEA <= 0.05): NA
</pre>

OpenMx version number: 2.13.2.95

**Comment by mcneale** [comment]
Why do you say it’s ill-fitting when chi-sq is zero?  Due to numerical error, the model fits a fraction better than the saturated model, hence the -3^-13 chi-sq.  Possibly, we should note this and suggest refitting ref models, but the difference is so tiny (less than a trillionth) that it could be set to zero.

If you follow the formulae for the fit indices, I think you’ll see that the answers are correct. This corner case with one variance & mean, whose independence model fits the same as the saturated, can hardly be expected to have very interesting fit indices.  Fits perfectly, enough said.

**Comment by tbates** [comment]
Sorry: pasted output of the  4-parameter model (parameters for each group), not the 2-parameter model.

The problem is that for the 2-parameter model (weight equated across the groups), `summary(m2)`  reports  TLI = 1,  and RMSEA:  0
That suggests (as you say) perfect fit, but it's wrong: the mean and variance of weight in the two groups both differ significantly.

Providing the refModels CHANGES, the reported values for TLI  and RMSEA

`summary(m2, refModels=m2Sat)` yields  TLI= NaN (not 1) and RMSEA a reasonable 0.56  [95% CI (0.32, 0.82)] instead of 0.

Maybe the issue here is that without refModels, the TLI and RMSEA should be being reported as NA, instead of 1 and 0 respectively.

**Comment by mhunter1** [comment]
We will add some checking for the theoretical bounds on our fit statistics.

**Comment by RMKirkpatrick** [comment]
I believe this is mostly resolved with 60f73e0.  However, when I run the attached script (adapted from @tbates ' syntax in this thread), I see CFIs of `-Inf`.  Is that a lingering concern--perhaps what @mhunter1 meant by "theoretical bounds"?
[issue221.R.txt](https://github.com/user-attachments/files/18171572/issue221.R.txt)

**Comment by RMKirkpatrick** [comment]
Good enough.
